src/render-html/js.js

```diff
@@ -180,7 +180,22 @@ const handleLongPress = e => {
   if (!lastTouchEventTimestamp || Date.now() - lastTouchEventTimestamp < 500) return;
 
   lastTouchEventTimestamp = 0;
-
+  if (e.target.matches('a')) {
+    sendMessage({
+      type: 'urlLongPress',
+      href: e.target.getAttribute('href'),
+      messageId: +getMessageIdFromNode(e.target),
+    });
+    return;
+  }
+  if (e.target.parentNode.matches('a')) {
+    sendMessage({
+      type: 'urlLongPress',
+      href: e.target.parentNode.getAttribute('href'),
+      messageId: +getMessageIdFromNode(e.target.parentNode),
+    });
+    return;
+  }
```

Comment:
Thanks -- this is an improvement over duplicating the logic from `handleLongPress` in the previous version!

These two blocks now have a lot in common -- I'd much prefer to see them de-duplicated.

Also, it's kind of a hack to be checking if the target element, or the target element's parent, is an `a` element. What if the target element's grandparent is the `a`, for example -- or the great-grandparent? This logic will be pretty brittle. Please try to find a cleaner way to identify if we're in a link.

Reply:
We're currently using this hack only for checking if we're in a link or not. I think a better logic is to be kept for another PR.

Reply:
Do something like this
https://github.com/zulip/zulip-mobile/blob/95676a3d4d56c37799fb51a19b75a4a0204458d2/src/render-html/js.js#L46

Reply:
This could be done, sure. But shall we not do this in a separate PR?